contracts/SinglePlayerCommit.sol

```diff
@@ -0,0 +1,339 @@
+/* SPDX-License-Identifier: MIT */
+pragma solidity ^0.6.10;
+pragma experimental ABIEncoderV2;
+
+import "@nomiclabs/buidler/console.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+//TODO Do we want console.log logging, start with Activity constructor?..
+
+contract SinglePlayerCommit is Ownable {
+    using SafeMath for uint256;
+
+    /******************
+    GLOBAL CONSTANTS
+    ******************/
+    IERC20 public token;
+    uint256 BIGGEST_NUMBER = uint256(-1);
+
+    /***************
+    DATA TYPES
+    ***************/
+    struct Measure {
+        string name;
+        bool allowed;
+    }
+
+    struct Activity {
+        string name; // e.g. "cycling"
+        bytes32[] measures; // keys from allowedMeasures
+        uint256[2][] ranges; // array of [min,max] goal values
+        address oracle;
+        bool allowed;
+    }
+
+    struct Commitment {
+        address committer; // user
+        bytes32 activity; // key from allowedActivities
+        bytes32 measure; // key from allowedMeasures
+        uint256 goalValue; // must be within range of Activity.measures[measureIndex]
+        uint256 start;
+        uint256 end;
+        uint256 stake; // amount of token staked, scaled by token decimals
+        bool exists; // flag to help check if commitment exists
+        uint256 reportedValue; // as reported by oracle
+        bool met; // whether the commitment has been met
+    }
+
+    /***************
+    EVENTS
+    ***************/
+    event NewCommitment(
+        address committer,
+        string activity,
+        string measure,
+        uint256 startTime,
+        uint256 endTime,
+        uint256 stake
+    );
+    event CommitmentEnded(address committer, bool met, uint256 amountPenalized);
+    event Deposit(address committer, uint256 amount);
+    event Withdrawal(address committer, uint256 amount);
+
+    /******************
+    INTERNAL ACCOUNTING
+    ******************/
+    mapping(bytes32 => Activity) public allowedActivities;
+    bytes32[] public activityList;
+
+    mapping(bytes32 => Measure) public allowedMeasures;
+    bytes32[] public measureList;
+
+    mapping(address => Commitment) public commitments; // active commitments
+    // address[] public committers; // addresses with active commitments
+
+    mapping(address => uint256) public balances; // current token balances
+    uint256 public committerBalance; // sum of current token balances
+
+    /********
+    FUNCTIONS
+    ********/
+    // constructor
+    constructor(
+        string memory _activity,
+        string[] memory _measures,
+        uint256[2][] memory _ranges,
+        address _oracle,
+        address _token
+    ) public {
+        // set up token interface
+        token = IERC20(_token);
+
+        // need to create fixed length bytes32 array to pass to _addActivity
```

Comment:
I think this may be the issue that's cause the measures to get missed. `_addActivity` takes a dynamic-sized array, not a fixed-length array.

Reply:
But I guess that doesn't explain why ranges are also getting missed. Those are passed to the constructor as dynamic-sized arrays and then directly to `_addActivity`, so I'm not sure what the issue is there